pynest/nest/lib/hl_api_sonata.py

```diff
@@ -0,0 +1,615 @@
+
+            # Simulate the network
+            sonata_net.Simulate()
+    """
+
+    def __init__(self, config, sim_config=None):
+
+        if not have_hdf5:
+            msg = ("SonataNetwork unavailable because NEST was compiled "
+                   "without HDF5 support")
+            raise kernel.NESTError(msg)
+        if not have_h5py:
+            msg = ("SonataNetwork unavailable because h5py is not installed "
+                   "or could not be imported")
+            raise kernel.NESTError(msg)
+
+        self._node_collections = {}
+        self._edges_maps = []
+        self._chunk_size_default = 2**20
+
+        self._is_nodes_created = False
+        self._is_network_built = False
+
+        self._conf = self._parse_config(config)
+        if sim_config is not None:
+            self._conf.update(self._parse_config(sim_config))
+
+        if self._conf["target_simulator"] != "NEST":
+            msg = "'target_simulator' in configuration file must be 'NEST'."
+            raise ValueError(msg)
+
+        if "dt" not in self._conf["run"]:
+            msg = ("Time resolution 'dt' must be specified in configuration "
+                   "file")
+            raise ValueError(msg)
+
+        SetKernelStatus({"resolution": self._conf["run"]["dt"]})
+
+    def _parse_config(self, config):
+        """Parse JSON configuration file.
+
+        Parse JSON config file and convert to a dictionary containing
+        absolute paths and simulation parameters.
+
+        Parameters
+        ----------
+        config : [str | pathlib.Path | pathlib.PurePath]
+            String or pathlib object describing the path to the JSON
+            configuration file.
+
+        Returns
+        -------
+        dict
+            SONATA config as dictionary
+        """
+
+        if not isinstance(config, (str, PurePath, Path)):
+            msg = ("Path to JSON configuration file must be passed as str, "
+                   "pathlib.PurePath or pathlib.Path")
+            raise TypeError(msg)
+
+        # Get absolute path
+        conf_path = Path(config).resolve(strict=True)
+        base_path = conf_path.parent
+
+        with open(conf_path) as fp:
+            conf = json.load(fp)
+
+        # Replace path variables (e.g. $MY_DIR) with absolute paths in manifest
+        for k, v in conf["manifest"].copy().items():
+            if "$BASE_DIR" in v:
+                v = v.replace("$BASE_DIR", ".")
+            conf["manifest"].update({k: base_path.joinpath(v).as_posix()})
+
+            if k.startswith("$"):
+                conf["manifest"][k[1:]] = conf["manifest"].pop(k)
+
+        def recursive_substitutions(config_obj):
+            # Recursive substitutions of path variables with entries from manifest
+            if isinstance(config_obj, dict):
+                return {k: recursive_substitutions(v) for k, v in config_obj.items()}
+            elif isinstance(config_obj, list):
```

Comment:
You can use the return-early pattern here, "else" is not necessary because all of the cases return from the function.